packages/next/src/server/lib/trace/tracer.ts

```diff
@@ -172,6 +193,13 @@ class NextTracerImpl implements NextTracer {
     return context
   }
 
+  public getTracePropagationData(): ClientTraceDataEntry[] {
+    const activeContext = context.active()
```

Comment:
In general, it is beneficial to propagate sampled and un-sampled contexts because telemetry systems often do traced-based sampling, where a tracing decision for a particular trace (usually decided on the first service) is propagated across the entire system, and the individual services of the system only report spans, if the incoming tracing decision is positive.

For example in the `traceparent` header there is a flag indicating whether the trace is sampled or not: https://www.w3.org/TR/trace-context/#sampled-flag If we didn't propagate the data when the trace is not sampled, the flag would not be needed in the first place.

Does that make sense? All in all this is good because it allows us to have less logic here. :)

Reply:
That's what I meant - the absence of the trace can be used to mean the traceflags are `00`. But I agree with the overall reasoning. You can keep this.